WcaOnRails/app/controllers/users_controller.rb

```diff
@@ -69,13 +69,13 @@ def select_nearby_delegate
 
   def edit_avatar_thumbnail
     @user = user_to_edit
-    return if redirect_if_cannot_edit_user(@user)
+    return if redirect_unless_user(:can_change_users_avatar?, @user)
```

Comment:
Thanks! Now that I understand it I have thoughts:

I think `redirect_unless_user(:can_change_users_avatar?, @user) and return` is the more common pattern. Also a bit weird, but my brain parses that without grinding to a halt.

I can see the attraction in writing code according to a pattern that you stick to even if it occasionally isn't needed. And also that it can serve as safety net in untested code. But I'm personally reluctant do either of those things. And here the exception seems to happen half the time.

An other source of my big confusion here was that the name `redirect_unless_user(...)` doesn't describe what it does. `redirect_to_root_unless_user(...)` would be a lot better.